File: backend/src/main/java/by/fpmibsu/findafriend/controller/controllers/AuthController.java
package by.fpmibsu.findafriend.controller.controllers;

import by.fpmibsu.findafriend.application.HandleResult;
import by.fpmibsu.findafriend.application.controller.*;
import by.fpmibsu.findafriend.application.mediatr.Mediatr;
import by.fpmibsu.findafriend.controller.commands.auth.SignInCommand;

@ControllerRoute(route = "/auth")
public class AuthController extends Controller {
    private final Mediatr mediatr;

    public AuthController(Mediatr mediatr) {
        this.mediatr = mediatr;
    }

    @Endpoint(path = "/signIn", method = HttpMethod.GET)
    public HandleResult signIn(@FromQuery(parameterName = "login") String login,
                               @FromQuery(parameterName = "password") String password) {
        return ok(mediatr.send(new SignInCommand(login, password)));
    }
}

File: BackendNET/Backend/Controllers/AuthController.cs
using Backend.Commands.Auth;
using MediatR;
using Microsoft.AspNetCore.Mvc;

namespace Backend.Controllers;

public class AuthController : ControllerBase
{
    private readonly IMediator _mediator;

    public AuthController(IMediator mediator)
    {
        _mediator = mediator;
    }

    [HttpPost("/auth/signIn")]
    public async Task<IActionResult> SignInAsync([FromQuery] string login, [FromQuery] string password)
    {
        return Ok(await _mediator.Send(new SignInCommand { Login = login, Password = password }));
    }
}